tools/syz-declextract/syz-declextract.cpp

```diff
@@ -18,31 +20,100 @@
 #include "llvm/Support/Casting.h"
 #include "llvm/Support/CommandLine.h"
 #include "llvm/Support/raw_ostream.h"
+#include <cstddef>
+#include <filesystem>
 #include <stdio.h>
 #include <string>
 #include <vector>
 
 using namespace clang;
 using namespace clang::ast_matchers;
 
+struct EnumData {
+  std::string name;
+  int value;
+  std::string file;
+};
+
 struct Param {
   std::string type;
   std::string name;
 };
 
-class Printer : public MatchFinder::MatchCallback {
+class EnumMatcher : public MatchFinder::MatchCallback {
+private:
+  std::vector<EnumData> EnumDetails;
+
+public:
+  std::vector<EnumData> getEnumData() { return EnumDetails; }
+  virtual void run(const MatchFinder::MatchResult &Result) override {
+    const auto *enumValue = Result.Nodes.getNodeAs<ConstantExpr>("enum_value");
+    if (!enumValue)
+      return;
+    const auto &name = enumValue->getEnumConstantDecl()->getNameAsString();
+    const auto value = int(*enumValue->getAPValueResult().getInt().getRawData());
+    const auto &path = std::filesystem::relative(
+        Result.SourceManager->getFilename(enumValue->getEnumConstantDecl()->getSourceRange().getBegin()).data());
+    EnumDetails.push_back({std::move(name), value, std::move(path)});
+  }
+};
+
+class Matcher : public MatchFinder::MatchCallback {
 private:
-  const std::string getSyzType(const Param &arg) { return "intptr"; }
+  const std::string nlaToSyz(const Expr *const policyType) {
+    // NOTE:This check is for when the policy is missing the field `type`.
+    // TODO:Gather information from other defined fields to better specify a type.
+    if (!policyType->getEnumConstantDecl()) {
+      return "intptr";
+    }
+    const auto type = policyType->getEnumConstantDecl()->getNameAsString().substr(4); // remove the NLA_ prefix
+    // Loosely based on https://elixir.bootlin.com/linux/v6.10/source/lib/nlattr.c
+    if (type == "U8" || type == "S8") {
+      return "int8";
+    }
+    if (type == "U16" || type == "S16" || type == "BINARY") {
```

Comment:
I think NLA_BINARY should translate to `array[int8]`.

If you was looking here:
https://elixir.bootlin.com/linux/v6.10.6/source/lib/nlattr.c#L134
I would assume .max in this case used as the max length of the blob, not max value of fixed-size int.

Reply:
Done!